File: Question 3 (Algorithm)/digital_calculator.py
def main():
    number1 = input("Input Number 1: ")
    number2 = input("Input Number 2: ")
    
    sum_of_numbers = calculate_sum(number1, number2)
    print("Sum of Numbers:", sum_of_numbers)
    
def calculate_sum(number1, number2):
    sum_list = []
    carry = 0
    max_length = max(len(number1), len(number2))
    
    for i in range(max_length):
        digit1 = get_digit(number1, i)
        digit2 = get_digit(number2, i)
        
        digit_sum = digit1 + digit2 + carry
        sum_list.append(str(digit_sum % 10))
        carry = digit_sum // 10
    
    if carry > 0:
        sum_list.append(str(carry))
    
    return ''.join(sum_list[::-1])
    
def get_digit(number, index):
    if index < len(number):
        return int(number[-1 - index])
    return 0

if _name_ == '_main_':
    main()

File: Question 3 (Algorithm)/digital_calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Input Number 1: ");
        String number1 = scanner.nextLine();
        System.out.print("Input Number 2: ");
        String number2 = scanner.nextLine();
        
        String sum = calculateSum(number1, number2);
        System.out.println("Sum of Numbers: " + sum);
    }
    
    public static String calculateSum(String number1, String number2) {
        StringBuilder sum = new StringBuilder();
        int carry = 0;
        int maxLength = Math.max(number1.length(), number2.length());
        
        for (int i = 0; i < maxLength; i++) {
            int digit1 = getDigit(number1, i);
            int digit2 = getDigit(number2, i);
            
            int digitSum = digit1 + digit2 + carry;
            sum.append(digitSum % 10);
            carry = digitSum / 10;
        }
        
        if (carry > 0) {
            sum.append(carry);
        }
        
        return sum.reverse().toString();
    }
    
    public static int getDigit(String number, int index) {
        if (index < number.length()) {
            return Character.getNumericValue(number.charAt(number.length() - 1 - index));
        }
        return 0;
    }
}
